src/datastructure/Image.cpp

```diff
@@ -419,14 +428,15 @@ FrameworkReturnCode Image::load(std::string imagePath)
     m_internalImpl = utils::make_shared<ImageInternal>();
     m_internalImpl->setData(pixels, spec.image_bytes(true));
     in->close();
+    delete[] pixels;
```

Comment:
Here you fixed a case, if everything went OK, but at line 402, for exemple, where there is a return statement, resulting in a memory leak (same thing if any exception occurs). You could put everything in a `try` block and put the `delete[]` in a finally, but we should avoid using `new`/`delete` as much as possible.
Solutions might be to either
* use a `std::unique_ptr<unsigned char[]>` (I think that might require a trick consisiting in explicitely give a deallocator, but I'm not sure).
[Here](https://en.cppreference.com/w/cpp/memory/unique_ptr/operator_at) a code snippet with an array of ints.

* use a `std::vector` you initialize with the `buffersize` capacity, and give access to openimageio to the internal array with `data`.

Reply:
It is a good idea to use std::unique_ptr<unsigned char[]> pixels(new unsigned char[size]).